modules/diskq/logqueue-disk.c

```diff
@@ -203,4 +225,61 @@ log_queue_disk_init_instance(LogQueueDisk *self, DiskQueueOptions *options, cons
   self->super.type = log_queue_disk_type;
 
   self->qdisk = qdisk_new(options, qdisk_file_id);
+  _set_serializer(self);
+}
+
+static gboolean
+_request_serializer(LogQueueDisk *self, SerializeArchive *sa, GError **error)
+{
+  gpointer args[] = {sa, error};
+  return qdisk_start_serialize(self->qdisk, args);
+}
+
+static gboolean
+_serialize_msg(LogQueueDisk *self, LogMessage *msg, SerializeArchive **sa)
+{
+  if (self->serialize_msg)
+    return self->serialize_msg(msg, *sa);
```

Comment:
if we passed self to serialize_msg at this point (as we usually do for virtual methods), the _set_serializer() would become moot and unnecessary, making the flow a bit easier to follow, IMHO.

That way the implementation of serialize_msg could either use the compaction flag or not depending on options->compaction. Am I missing something?

Reply:
`self->serialize_msg` has been removed from this PR.